pkg/driver/driver_alicloud.go

```diff
@@ -138,43 +83,33 @@ func (c *AlicloudDriver) Create() (string, string, error) {
 
 	if c.AlicloudMachineClass.Spec.SystemDisk != nil {
 		request.SystemDiskCategory = c.AlicloudMachineClass.Spec.SystemDisk.Category
-		request.SystemDiskSize = requests.NewInteger(c.AlicloudMachineClass.Spec.SystemDisk.Size)
+		request.SystemDiskSize = fmt.Sprintf("%d", c.AlicloudMachineClass.Spec.SystemDisk.Size)
 	}
 
-	for key, value := range c.AlicloudMachineClass.Spec.Tags {
-		if strings.Contains(key, "kubernetes.io/cluster/") {
-			request.Tag1Key = key
-			request.Tag1Value = value
-		} else if strings.Contains(key, "kubernetes.io/role/") {
-			request.Tag2Key = key
-			request.Tag2Value = value
-		} else if request.Tag3Key == "" {
-			request.Tag3Key = key
-			request.Tag3Value = value
-		} else if request.Tag4Key == "" {
-			request.Tag4Key = key
-			request.Tag4Value = value
-		} else if request.Tag5Key == "" {
-			request.Tag5Key = key
-			request.Tag5Value = value
-		} else if request.Tag6Key == "" {
-			request.Tag6Key = key
-			request.Tag6Value = value
-		} else if request.Tag7Key == "" {
-			request.Tag7Key = key
-			request.Tag7Value = value
-		} else if request.Tag8Key == "" {
-			request.Tag8Key = key
-			request.Tag8Value = value
-		} else if request.Tag9Key == "" {
-			request.Tag9Key = key
-			request.Tag9Value = value
-		} else if request.Tag10Key == "" {
-			request.Tag10Key = key
-			request.Tag10Value = value
+	hasCluster := false
+	hasRole := false
+
+	tags := []ecs.RunInstancesTag{{}, {}}
+	for k, v := range c.AlicloudMachineClass.Spec.Tags {
+		if strings.Contains(k, "kubernetes.io/cluster/") {
+			hasCluster = true
+			tags[0].Key = k
```

Comment:
1. Please be notified **tags** has already been initialized with ```[]ecs.RunInstancesTag{{}, {}}```
2. Role and cluster tags should always exist, otherwise, an error will be returned. If duplicated tags exit, the latter always overrites the previous

Reply:
Hi @hardikdr , 
I added a unit test.
And also, I created a machine class with disordered tags and the tags in the real VM are expected.

Reply:
Thank you so much.